src/us/kbase/workspace/test/workspace/WorkspaceUnitTest.java

```diff
@@ -516,4 +531,169 @@ private void failGetObjects(
 			TestCommon.assertExceptionCorrect(got, expected);
 		}
 	}
+	
+	private final String SAVE_OBJECT_FAIL_VAL_TYPE_STRING = "Mod.Type-3.2";
+	private final int SAVE_OBJECT_FAIL_VAL_MAX_IDS = 602214;
+	private final UObject SAVE_OBJECT_FAIL_GOOD_DATA = new UObject(ImmutableMap.of("a", "b"));
+	private final UObject SAVE_OBJECT_FAIL_BAD_DATA = new UObject(ImmutableMap.of("c", "d"));
+	private final WorkspaceSaveObject SAVE_OBJECT_FAIL_MEDIOCRE_OBJECT = new WorkspaceSaveObject(
+			new ObjectIDNoWSNoVer(67),
+			SAVE_OBJECT_FAIL_GOOD_DATA,
+			TypeDefId.fromTypeString(SAVE_OBJECT_FAIL_VAL_TYPE_STRING),
+			new WorkspaceUserMetadata(),
+			Provenance.getBuilder(new WorkspaceUser("u"), now()).build(),
+			false);
+	
+	private WorkspaceSaveObject saveObjectValidationFailGetBadObject(final ObjectIDNoWSNoVer id) {
+		return new WorkspaceSaveObject(
+				id,
+				SAVE_OBJECT_FAIL_BAD_DATA,
+				TypeDefId.fromTypeString(SAVE_OBJECT_FAIL_VAL_TYPE_STRING),
+				new WorkspaceUserMetadata(),
+				Provenance.getBuilder(new WorkspaceUser("u"), now()).build(),
+				false);
+	}
+	@Test
+
+	public void saveObjectValidationFailNoSuchModuleException() throws Exception {
```

Comment:
This messed up spacing is fixed locally and will be in a future PR

Reply:
can you set up a code linter/formatter to run as a pre-commit hook so you don't have to worry about this kind of thing?

Reply:
Hmm, I'm not really a huge fan of style linters - 99% of the stuff they catch is trivial and not worth worrying about. There's just too much noise, and the number of times linters have actually helped me is tiny.

These cases are rare enough and harmless enough that I don't think the trouble is worth it. Maybe my perspective will change in the future, but for right now I think the cure is worse than the disease.